Added repeat state and service for media_players mpd, universal and demo

todschmidt opened the pull request:
## Description:

Added repeat state and service for media_players mpd, universal and demo

**Related issue (if applicable):** fixes # N/A

**Pull request in https://github.com/home-assistant/home-assistant** home-assistant/home-assistant.github.io#

## Example entry for `configuration.yaml` (if applicable):
```yaml

```

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in https://github.com/home-assistant/home-assistant.github.io

If the code communicates with devices, web services, or third-party tools:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

molobrakos committed:
Upgrade tellduslive library version (closes https://github.com/home-assistant/home-assistant/issues/10922) (#10950)

mateuszdrab committed:
Fix linksys_ap.py by inheriting DeviceScanner (#10947)

As per issue #8638, the class wasn't inheriting from DeviceScanner, this commit patches it up.

todschmidt committed:
Added repeat to media player and mpd

todschmidt committed:
Added repeat to media player and mpd

todschmidt committed:
Fixed version numbers

todschmidt committed:
Fixed version numbers

todschmidt committed:
Added repeat to these players as well

todschmidt committed:
Added repeat to these players as well

todschmidt committed:
Updated test

todschmidt committed:
Updated test

homeassistant commented:
Hi @tschmidty69, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#10960).

Once you do that we will be able to review and accept this pull request.

Thanks!

todschmidt committed:
Added gi to gstreamer requirements

todschmidt committed:
Added gi to gstreamer requirements

homeassistant commented:
Hi @tschmidty69, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#10960).

Once you do that we will be able to review and accept this pull request.

Thanks!

todschmidt committed:
Fixed whitespace and line too long errors

todschmidt committed:
Fixed whitespace and line too long errors

todschmidt committed:
Stop rushing to get dinner...

todschmidt committed:
Stop rushing to get dinner...

todschmidt committed:
Update __init__.py

todschmidt committed:
Update __init__.py

todschmidt committed:
Update demo.py

todschmidt committed:
Update demo.py

todschmidt committed:
Update test_universal.py

todschmidt committed:
Update test_universal.py

todschmidt committed:
Update demo.py

todschmidt committed:
Update demo.py

Commit:
Add ADS component (#10142)

* add ads hub, light and switch

* add binary sensor prototype

* switch: use adsvar for connection

* fix some issues with binary sensor

* fix binary sensor

* fix all platforms

* use latest pyads

* fixed error with multiple binary sensors

* add sensor

* add ads sensor

* clean up after shutdown

* ads component with platforms switch, binary_sensor, light, sensor

add locking

poll sensors at startup

update state of ads switch and light

update ads requirements

remove update() from constructors on ads platforms

omit ads coverage

ads catch read error when polling

* add ads service

* add default settings for use_notify and poll_interval

* fix too long line

* Fix style issues

* no pydocstyle errors

* Send and receive native brightness data to ADS device to prevent issues with math.floor reducing brightness -1 at every switch

* Enable non dimmable lights

* remove setting of self._state in switch

* remove polling

* Revert "remove polling"

This reverts commit 7da420f82385a4a5c66a929af7025c00ed197e86.

* add service schema, add links to documentation

* fix naming, cleanup

* re-remove polling

* use async_added_to_hass for setup of callbacks

* fix comment.

* add callbacks for changed values

* use async_add_job for creating device notifications

* set should_poll to False for all platforms

* change should_poll to property

* add service description to services.yaml

* add for brigthness not being None

* put ads component in package

* Remove whitespace

* omit ads package

craigjmidwinter committed:
Reload closest store on api menu request (#10962)

* reload closest store on api request

* revert change from debugging

craigjmidwinter committed:
Reload closest store on api menu request (#10962)

* reload closest store on api request

* revert change from debugging

HerrHofrat committed:
Gearbest sensor (#10556)

* Added Gearbest Sensor

* Updated required files

* Fixed houndci-bout findings

* Fix tox lint errors

* Changed code according to review
Implemented library version 1.0.5

* Fixed houndci-bot findings

* Fixed tox lint issues

* Updated item schema to use has_at_least_one_key
Added conf constants

* Remove CONF_ constants and import them from homeassistant.const

* Removed CurrencyConverter from hass
Fixed couple of issues found by MartinHjelmare

b10m committed:
Add Ziggo Mediabox XL media_player (#10514)

* Add Ziggo Mediabox XL media_player

* Using pypi module ziggo-mediabox-xl now.

* Code review changes

ziotibia81 committed:
Generic thermostat initial_operation_mode (#10690)

* Generic thermostat restore operation mode

* Test restore operation mode

* Fix trailing whitespace

* Fix line too long

* Fix test duplicate entity_id

* Fix test

* async_added_to_hass modify modify internal state

* Test inital_operation_mode

* More restore state tests

* Fix whitespace

* fix test_custom_setup_param

* Test "None" target temp

emlove committed:
Use new build path for dev translations (#10937)

masarliev committed:
Add option to set default hide if away for new devices (#10762)

* Option to change hide_if_away

* tests fix

* change new device defaults

* tests and requested changes

* fix assert

Commit:
Allow chime to work for wink siren/chime (#10961)

* Allow Wink siren/chimes to work

* Updated requirements_all.txt

Commit:
Allow chime to work for wink siren/chime (#10961)

* Allow Wink siren/chimes to work

* Updated requirements_all.txt

balloob committed:
Revert pychromecast update (#10989)

* Revert pychromecast update

* Update cast.py

balloob committed:
Revert pychromecast update (#10989)

* Revert pychromecast update

* Update cast.py

andrey-git committed:
Require FF43 for latest js (#10941)

* Require FF43 for latest js

`Array.prototype.includes` added in Firefox 43

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

* Update __init__.py

andrey-git committed:
Require FF43 for latest js (#10941)

* Require FF43 for latest js

`Array.prototype.includes` added in Firefox 43

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

* Update __init__.py

balloob committed:
Version bump to 0.59.2

balloob committed:
Merge pull request #10990 from home-assistant/release-0-59-2

0.59.2

balloob committed:
Merge branch 'master' into dev

DanNixon committed:
Allow disabling the LEDs on TP-Link smart plugs (#10980)

balloob committed:
Update frontend to 20171206.0

masarliev committed:
Meraki AP Device tracker (#10971)

* Device tracker for meraki AP

* styles fix

* fix again

* again

* and again :)

* fix hide if away

* docs and optimization

* tests and fixes

* styles

* styles

* styles

* styles

* styles fix. Hope last

* clear track new

* changes

* fix accuracy error and requested changes

* remove meraki from .coveragerc

* tests and minor changes

* remove location

pvizeli committed:
Update tradfri.py (#10991)

todschmidt committed:
Added repeat to spotify and kodi to match docs

todschmidt committed:
Added repeat to spotify and kodi to match docs

todschmidt committed:
Merge branch 'dev' of https://github.com/tschmidty69/home-assistant into dev

tschmidty69 commented:
Updated docs to add service and added kodi and spotify support

OddBloke committed:
webostv: Ensure source exists before use (#10959)

In a case where either (a) an incorrect source name is used, or (b) the
TV isn't currently queryable (e.g. it's off), we get tracebacks because
we assume the source that we are being asked to select exists in
self._source_list.

This makes the lookup code a little more rugged, and adds in a warning
message (in place of the current exception).

Commit:
Ensure Docker script files uses LF line endings to support Docker for Windows. (#10067)

todschmidt committed:
Update spotify.py

todschmidt committed:
Update spotify.py

alanfischer committed:
Added Vera scenes (#10424)

* Added Vera scenes

* Fixed flake8 issues

* Fixed comments

* Moved vera to use hass.data

* Made requested changes

jeroenterheerdt committed:
Fix Egardia alarm status shown as unknown after restart (#11010)

lwis committed:
Handle OSError when forcibly turning off media_player.samsungtv (#10997)

todschmidt committed:
Snips.py will use rawValue instead of value if no value present in slot['value']

todschmidt committed:
Snips.py will use rawValue instead of value if no value present in slot['value']

todschmidt committed:
Merge branch 'dev' of github.com:tschmidty69/home-assistant into dev

todschmidt committed:
Snips will use rawValue instead of value if value not present in slot

todschmidt committed:
Snips will use rawValue instead of value if value not present in slot

mar-schmidt committed:
Shuffle support in Sonos (#10875)

* initial commit of shuffle option for sonos

* added test

* Small adjustments to adhere to review requests

* Removed unnessesary setting of variable. Use shuffle state from soco instead

todschmidt committed:
Fixed whitespace issues

todschmidt committed:
Merge branch 'dev' of github.com:tschmidty69/home-assistant into dev

todschmidt committed:
Update snips component to use the hermes/intent topic per snips 2.0
